cmd/prometheus/main.go

```diff
@@ -265,36 +267,46 @@ func main() {
 	// RoutePrefix must always be at least '/'.
 	cfg.web.RoutePrefix = "/" + strings.Trim(cfg.web.RoutePrefix, "/")
 
-	cfg.tsdb.RetentionDuration = chooseRetention(oldFlagRetentionDuration, newFlagRetentionDuration)
+	{ // Time retention settings.
+		if oldFlagRetentionDuration != 0 {
+			level.Warn(logger).Log("deprecation_notice", "'storage.tsdb.retention' flag is deprecated use 'storage.tsdb.retention.time' instead.")
+			cfg.tsdb.RetentionDuration = oldFlagRetentionDuration
+		}
 
-	// Check for overflows. This limits our max retention to ~292.5y.
-	if cfg.tsdb.RetentionDuration < 0 {
-		cfg.tsdb.RetentionDuration = math.MaxInt64
-	}
+		// When the new flag is set it takes precedence.
+		if newFlagRetentionDuration != 0 {
+			cfg.tsdb.RetentionDuration = newFlagRetentionDuration
+		}
 
-	if cfg.tsdb.MaxBlockDuration == 0 {
-		cfg.tsdb.MaxBlockDuration = cfg.tsdb.RetentionDuration / 10
+		if cfg.tsdb.RetentionDuration == 0 && cfg.tsdb.MaxBytes == 0 {
+			cfg.tsdb.RetentionDuration = defaultRetentionDuration
+			level.Warn(logger).Log("msg", "no time or size retention was set so using the default time retention", "duration", defaultRetentionDuration)
```

Comment:
IMO it shouldn't a warning. A debug message is probably enough.

Reply:
I'd suggest Info, and mentioning what the default is. We've had quite a few user questions around this over the years.

Reply:
This will only appear when no time or size retention is set and I prefer to be more distinct than just an info message so I would rather keep it as a warning.

@brian-brazil  	the default value is already in the message.

Reply:
As a user, I find it confusing to get a warning log while I just run `./prometheus`

Reply:
Confusing in what way?

for me a `warn` is something out of the ordinary. Something you should pay attention to. 
In this case since no explicit flag or a setting was set a warning is raised to notify that a default of 15d retention is automatically applied. Without this it is easy to miss along the other info messages.

Reply:
Because the warning wasn't present in previous versions and we aren't changing anything to the way retention settings work. I assume that a significant fraction of users run Prometheus without specifying the retention period and they might be confused seeing a warning log.

Reply:
well there is a change in the behaviour. Before this was a default and now it is only applied when no time/size retention is set. 


Anyway it is not that important so lets have one more person give an opinion and will leave or change accordingly.

Reply:
I vote for INFO.

Reply:
updated, thanks for the feedback.

Reply:
Thanks Krasi!